src/training/main.py

```diff
@@ -267,7 +267,8 @@ def main(args):
         if is_master(args):
             logging.info(f'Start epoch {epoch}')
 
-        train_one_epoch(model, data, epoch, optimizer, scaler, scheduler, args, writer)
+        loss = create_loss(args)
```

Comment:
we shouldn't have to create the loss every epoch but this isn't very bad. It does raise an interesting question though - for methods like FLAVA there's this notion of task sampling where you sometimes train one loss and sometimes other loss, maybe we should think about how this mechanism would work. Maybe you wouldn't swap every epoch but more finely within the epoch. Anyway, for this maybe just check for training data and if that exists create the loss, otherwise set it to None